tests/unit/test_stream.py

```diff
@@ -77,6 +81,24 @@ def test_snap(self):
         market_books = self.stream.snap(['1.1'])
         assert market_books == [mock_cache.create_resource()]
 
+        self.stream._caches = {'1.{}'.format(i): mock_cache
```

Comment:
how does this work that it does not replace L73?

Reply:
It does replace L73.  This is to force a `RuntimeError: dictionary changed size during iteration` during the `.snap()` call.  Problem is, you need a "big" dict to force this to happen (generally).  You need the size of the dict to change during iteration, which is what is happening in the thread.